pkg/controller/render/render_controller_test.go

```diff
@@ -285,12 +287,21 @@ func TestIgnValidationGenerateRenderedMachineConfig(t *testing.T) {
 
 	// verify that an invalid igntion config (here a config with content and an empty version,
 	// will fail validation
-	mcs[1].Spec.Config.Ignition.Version = ""
+	ignCfg, _, err := ign.Parse(mcs[1].Spec.Config.Raw)
+	require.Nil(t, err)
```

Comment:
> One more thing to note that is we may need to somewhat modify https://github.com/openshift/machine-config-operator/blob/master/manifests/machineconfig.crd.yaml#L67,
> Although that said since we are not changing UX, I suppose its fine to still list is as an ignition (as opposed to rawExtension) object

AFAICT the only change for Spec3 here would be to remove the `filesystem` field inside dirs and files, but that would make it incompatible with Spec2. I therefore believe it should be safe to leave as-is.

Reply:
keeping the current style here